### PR TITLE
Lizmap service etag

### PR DESCRIPTION
**Pull request by rldhont**
Add the [ETag HTTP header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag) for some service responses with `Cache-Control: no-cache` header.

The responses for GetProjectConfig, GetCapabilities and GetProj4 have an Etag HTTP header

* Funded by 3liz

### CODE REVIEW

**Review by laurentj — APPROVED** [review]
lgtm

File: lizmap/modules/lizmap/controllers/service.classic.php [hunk 1]
```diff
@@ -774,14 +820,32 @@ protected function GetProj4()
...
+        $etag .= '-'.$authid;
+        $cacheHandler = $this->project->getCacheHandler();
+        $etag .= '-'.$cacheHandler->getFileTime().'~'.$cacheHandler->getCfgFileTime();
+        $etag = sha1($etag);
+        if ($rep->isValidCache(null, $etag)) {
+            return $rep;
+        }
+
+        // Get content
+        $content = $this->project->getProj4($authid);
```
[the remark this answers is not in the document]
Comment (rldhont) [reply lizmap/modules/lizmap/controllers/service.classic.php:841]: It is read from QGIS project.

**Comment by rldhont** [comment]
@laurentj I have introduced a new class to clean the code, OGCResponse. Would you prefer that I separate it in an other PR ?

**Comment by rldhont** [comment]
@nboisteault @mdouchin new review ?

**Review by laurentj — APPROVED** [review]
lgtm

**Pull request merged**